libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -399,6 +399,11 @@ void AP_GPS::detect_instance(uint8_t instance)
     struct detect_state *dstate = &detect_state[instance];
     uint32_t now = AP_HAL::millis();
 
+    state[instance].instance = instance;
```

Comment:
The one comment I'd make (I sent Fransisco this offline) is that There is no value in setting this in the detect_instance loop, as state[instance].instance will always be the same value. We should be setting this in the AP_GPS constructor. (We're just wasting time in detect_instance doing it here)